lib/wasix/src/runtime/module_cache/types.rs

```diff
@@ -123,32 +124,36 @@ impl CacheError {
 
 /// The SHA-256 hash of a WebAssembly module.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct ModuleHash([u8; 32]);
+pub struct ModuleHash([u8; 8]);
 
 impl ModuleHash {
     /// Create a new [`ModuleHash`] from the raw SHA-256 hash.
-    pub fn from_bytes(key: [u8; 32]) -> Self {
+    pub fn from_bytes(key: [u8; 8]) -> Self {
         ModuleHash(key)
     }
 
     /// Parse a sha256 hash from a hex-encoded string.
     pub fn parse_hex(hex_str: &str) -> Result<Self, hex::FromHexError> {
-        let mut hash = [0_u8; 32];
+        let mut hash = [0_u8; 8];
         hex::decode_to_slice(hex_str, &mut hash)?;
         Ok(Self(hash))
     }
 
     /// Generate a new [`ModuleCache`] based on the SHA-256 hash of some bytes.
     pub fn sha256(wasm: impl AsRef<[u8]>) -> Self {
+        Self::xxhash(wasm)
```

Comment:
Having sha256 use xxhash is wrong. We should change the usage

Reply:
I know. It's a partial implementation. Instead of having to hunt down every `sha256` usage, this allowed me to test where in our pipeline I'm going to encounter issues. I'll rename the function hence:

> If all tests pass, my next commit will fully migrate to xxhash.